coordinator/cluster/cluster.go

```diff
@@ -20,17 +20,40 @@ import (
 // LoadAgentCluster is the object holding information about all the load-test
 // agents available in the cluster.
 type LoadAgentCluster struct {
-	config LoadAgentClusterConfig
-	agents []*client.Agent
-	errMap map[*client.Agent]errorTrack
-	log    *mlog.Logger
+	config   LoadAgentClusterConfig
+	ltConfig loadtest.Config
+	agents   []*client.Agent
+	errMap   map[*client.Agent]errorTrack
+	log      *mlog.Logger
 }
 
 type errorTrack struct {
 	lastError   int64
 	totalErrors int64
 }
 
+func createAgent(agent *client.Agent, ltConfig loadtest.Config) error {
+	// TODO: UserController config should probably come from the upper layer
+	// and be passed through.
+	var ucConfig interface{}
+	var err error
+	switch ltConfig.UserControllerConfiguration.Type {
+	case loadtest.UserControllerSimple:
+		ucConfig, err = simplecontroller.ReadConfig("")
+	case loadtest.UserControllerSimulative:
+		ucConfig, err = simulcontroller.ReadConfig("")
+	}
```

Comment:
Would a default case be necessary here?

Reply:
It's not strictly necessary. It would be a way to invalidate early essentially but I didn't want to add more logic. 
If the config is bad the `Create()` API call will fail with the reason why.